xapian-applications/omega/index_file.cc

```diff
@@ -545,6 +545,77 @@ index_add_document(const string & urlterm, time_t last_altered,
     }
 }
 
+bool
+index_update_entry(const string & urlterm,
+		   DirectoryIterator & d,
+		   Xapian::docid & did)
+{
+    if (!did) {
+	Xapian::PostingIterator p = db.postlist_begin(urlterm);
+	if (p != db.postlist_end(urlterm))
+	    did = *p;
+    }
+
+    if (did) {
```

Comment:
I would suggest dealing with the "no docid" case returning `false` here - then you can avoid having the whole of the actual implementation indented by an extra level.

Reply:
Yes, you are are right. I didn't realize of that. I will change it